doc/variableexplorer.rst

```diff
@@ -2,95 +2,235 @@
+
+
+======================
+Advanced Functionality
+======================
+
+The context menu, available by right-clicking any variable, provides numerous additional options to interact with objects of various types. These include renaming, removing or editing existing variables, as well as the :guilabel:`duplicate` option to create a new copy of one of them under a new name you enter in the resulting dialog box.
+
+.. image:: images/variable_explorer/variable-explorer-duplicate.gif
+   :align: center
+   :alt: Variable Explorer showing duplicating a variable
+
+| 
+
+Furthermore, you can copy and paste the value of a variable, saving it in the Variable Explorer with any name that you choose. This allows you to change the type of the variable that you are pasting which can be very useful, allowing to, for example, easily copy the elements of a list into an array.
+
+.. image:: images/variable_explorer/variable-explorer-copy-paste.gif
+   :align: center
+   :alt: Variable Explorer showing copying list into array
+
+| 
 
-* Integers
-* Floats
-* Complex numbers
-* Strings
-* ``datetime`` dates and ``Timedelta`` s
-* Lists
-* Tuples
-* Sets
-* Dictionaries
-* NumPy arrays and matrices
-* Pandas ``DataFrame``, ``TimeSeries`` and ``Index`` (any type) objects
-* ``PIL``/``Pillow`` images
-* Namespaces
+Additionally, you can create a variable from scratch directly in the Variable Explorer with the :guilabel:`Insert` option, which allows you to type the key (which should be in quotation marks) and the value for the item that you want to insert. In addition to adding a new top-level variable, this feature also allows you to create a new key in a dictionary, a new element in a list, and much more.
 
+.. image:: images/variable_explorer/variable-explorer-insert.gif
+   :align: center
+   :alt: Variable Explorer showing insertion of a new variable
+
+| 
+
+For lists and Numpy arrays, more advanced options are available, including generating plots and histograms of their values appropriate to their type and dimensions.
+
+.. image:: images/variable_explorer/variable-explorer-histogram-plot.gif
+   :align: center
+   :alt: Plot window showing a plot, generated via the previous options
 
+|
+
+You can even save an array to a ``.npy`` file by simply clicking the appropriate option, which can later be loaded by Spyder or in your code via `numpy.load()`.
+
+.. image:: images/variable_explorer/variable-explorer-contextmenu-array.png
+   :width: 500
+   :align: center
+   :alt: Context menu for an int array, with the Show image option selected
+
+|
+
+For two-dimensional arrays, you can also display them as images, treating their values as RGB colors. For this, Spyder uses Matplotlib's colormaps, which can be `easily changed to match your preferences`_.
+
+.. _easily changed to match your preferences: https://matplotlib.org/3.1.0/tutorials/colors/colormaps.html
+
+.. image:: images/variable_explorer/variable-explorer-show-image.gif
```

Comment:
Same here: I suggest to show only the Variable Explorer (left) and Plots (right) on the gif.

Reply:
Done

Reply:
LGTM